userspace/libscap/scap.c

```diff
@@ -1169,6 +1169,62 @@ int32_t scap_get_stats(scap_t* handle, OUT scap_stats* stats)
 	return SCAP_SUCCESS;
 }
 
+int scap_get_modifies_state_ppm_sc(uint32_t* ppm_sc_array)
+{
+	if(ppm_sc_array == NULL)
+	{
+		return SCAP_FAILURE;
+	}
+
+#ifdef __linux__
+	// Collect EF_MODIFIES_STATE events
+	for (int i = 0; i < PPM_EVENT_MAX; i++)
+	{
+		if (g_event_info[i].flags & EF_MODIFIES_STATE)
+		{
+			for (int j = 0; j < SYSCALL_TABLE_SIZE; j++)
+			{
+				if (g_syscall_table[j].exit_event_type == i || g_syscall_table[j].enter_event_type == i)
+				{
+					uint32_t ppm_sc_code = g_syscall_code_routing_table[i];
+					ppm_sc_array[ppm_sc_code] = 1;
+				}
+			}
+		}
+	}
+
+	// Collect UF_NEVER_DROP syscalls
+	for (int j = 0; j < SYSCALL_TABLE_SIZE; j++)
+	{
+		if (g_syscall_table[j].flags & UF_NEVER_DROP)
```

Comment:
@FedeDP would you be amenable to consider using a separate new flag instead of `UF_NEVER_DROP` similarly to how you did it before with `UF_SIMPLEDRIVER_KEEP`? Maybe something like `UF_ENFORCED_SYSCALLS_INTEREST`, name does not matter as much as I think the separation of usages and purposes of flags do.

Reply:
I think the correct flag is already `EF_MODIFIES_STATE` IMHO ;) but we want to be conservative and also add the `UF_NEVER_DROP` for the moment being. We just need to check that EF_MODIFIES_STATE events and UF_NEVER_DROP syscalls basically match 1:1 :)

Reply:
(They do not). But again, this sinsp API just returns the minimum set of syscalls that are needed by libs, therefore i think that using both flags is good!

Reply:
Maybe in the future, we can introduce a new flag but before doing that we need to clarify and clean the actual flags as I wrote here https://github.com/falcosecurity/libs/pull/521#discussion_r957147904 WDYT?

Reply:
@FedeDP and @Andreagit97 agreed too risky to break things. Would be up to an exercise to audit everything and come up with a good knowledge base to be able to support Falco rules settings that only use few syscalls without breaking the state or integrity of sinsp ... this probably goes in hand with some Threat Modeling audits. Maybe this PR should focus on setting up all APIs and follow up PRs can take care of additional optimizations which was the plan all along I suppose.